Jenkinsfile

```diff
@@ -0,0 +1,110 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+/*
+ * This Jenkinsfile is intended to run on https://jenkins.ci.cloudbees.com and may fail anywhere else.
+ * It makes assumptions about plugins being installed, labels mapping to nodes that can build what is needed, etc.
+ *
+ * The required label is "hi-speed" */
+
+// TODO: Figure out how to run just changed/affected-by-changes tests when running from a PR.
+// TODO: Get https://issues.jenkins-ci.org/browse/JENKINS-33274 fixed and configured so that we don't have to run
+//   serially for the first run on a branch.
+
+timestamps {
+    // Only keep the 50 most recent builds.
+    properties([[$class: 'jenkins.model.BuildDiscarderProperty', strategy: [$class: 'LogRotator',
+                                                                            numToKeepStr: '50']]])
+
+    stage "Load libs"
+    loadLibs { functions, ath ->
+
+        stage "Fetch WAR and ATH source"
+        parallel 'jenkins-war': { functions.stashJenkinsWar(war, ath.athLabel(), true) },
+            'get-ath': { ath.stashAth(functions) },
+            failFast: true
+
+        stage "Run tests"
+        def branches = ath.getAthBranches(
+            functions,       // Used for things like getting the Maven environment set up.
+            ath.athLabel(),  // What label to use for execution.
+            false,           // Whether to archive JUnit report files for processing in a later job.
+            3,               // Number of times to retry failed tests
+            4                // Parallel branch count, in addition to the cucumber branch
+        )
+
+        parallel branches
+    }
+}
+
+/**
+ * Loads the libraries using known parameters for repo/branch/node if they're specified, and then makes the libraries
+ *   available in the execution of the body.
+ *
+ * @param body Closure to execute.
+ */
+def loadLibs(def body) {
+    String functionsRepo = defaultValueIfNotSet("FUNCTIONS_REPO", "jenkinsci/jenkins-project-pipeline-lib")
```

Comment:
Anyway the _Replay_ function lets you play with library changes more easily than what I am guessing this hook was for.